pkg/controller/cache/cache_controller.go

```diff
@@ -92,10 +111,163 @@ func (r *ReconcileCache) Reconcile(request reconcile.Request) (reconcile.Result,
+	listOps := &client.ListOptions{Namespace: ispnInstance.GetClusterName(), LabelSelector: labelSelector}
+	err = r.client.List(context.TODO(), podList, listOps)
+	if err != nil || (len(podList.Items) == 0) {
+		reqLogger.Error(err, "failed to list pods", "Infinispan.Namespace")
+		return reconcile.Result{}, err
+	}
+	cacheName := instance.Name
+	if instance.Spec.Name != "" {
+		cacheName = instance.Spec.Name
+	}
+	caches, err := cluster.CacheNames(user, pass, podList.Items[0].Name, instance.Namespace, ispnctrl.Protocol(ispnInstance))
+	if err == nil {
+		for i, cn := range caches {
+			if cn == cacheName {
+				reqLogger.Info("Cache already exists")
+				// Check if template matches?
+				break
+			}
+			if i == len(caches)-1 {
+				reqLogger.Info("Cache doesn't exist, create it")
+				podName := podList.Items[0].Name
+				xmlTemplate := instance.Spec.Template
+				if xmlTemplate == "" {
+					xmlTemplate, err = ispnctrl.GetDefaultCacheTemplateXML(podName, ispnInstance, cluster, reqLogger)
+				}
+				if err != nil {
+					reqLogger.Error(err, "Error getting default XML")
+					return reconcile.Result{}, err
+				}
+				reqLogger.Info(xmlTemplate)
+				err = cluster.CreateCacheWithAuth(user, pass, instance.Spec.Name, xmlTemplate, podName, instance.Namespace, ispnctrl.Protocol(ispnInstance))
+				if err != nil {
+					reqLogger.Error(err, "Error in creating cache")
+					return reconcile.Result{}, err
+				}
+			}
+		}
+	} else {
+		reqLogger.Error(err, "Error getting caches list")
+		return reconcile.Result{}, err
+	}
+
+	reqLogger.Info("Update CR status with connection info")
+
+	// Search the service associated to the cluster
+	serviceList := &corev1.ServiceList{}
+	labelSelector = labels.SelectorFromSet(ispnutil.GetLabelsForInfinispan(ispnInstance.Name, "infinispan-service"))
+	listOps = &client.ListOptions{Namespace: ispnInstance.Namespace, LabelSelector: labelSelector}
+	err = r.client.List(context.TODO(), serviceList, listOps)
+	if err != nil {
+		reqLogger.Error(err, "failed to select cluster service")
+		return reconcile.Result{}, err
+	}
+	if len(serviceList.Items) != 1 {
+		err = fmt.Errorf("Zero or more than 1 service associated to the cluster")
+		reqLogger.Error(err, fmt.Sprintf("Size of the result list: %d", len(serviceList.Items)))
+		return reconcile.Result{}, err
+	}
 
+	statusUpdate := false
+	if instance.Status.ServiceName != serviceList.Items[0].Name {
+		instance.Status.ServiceName = serviceList.Items[0].Name
+		statusUpdate = true
+	}
+	statusUpdate = statusUpdate || setCondition(instance, "Ready", "True", "")
+	if statusUpdate {
+		err = r.client.Status().Update(context.TODO(), instance)
+		if err != nil {
+			reqLogger.Error(err, "Unable to update Cache "+instance.Name+" status")
+			return reconcile.Result{}, err
+		}
+	}
 	return reconcile.Result{}, nil
 }
+
+// SetCondition set condition to status
+func setCondition(cache *v2alpha1.Cache, condition string, status metav1.ConditionStatus, message string) bool {
```

Comment:
Looks like methods like this may be on top of the v2alpha1.Cache entity?

Reply:
ok ,will do

Reply:
done